app/uk/gov/hmrc/tradergoodsprofiles/controllers/RemoveRecordController.scala

```diff
@@ -43,40 +41,45 @@ class RemoveRecordController @Inject() (
   cc: ControllerComponents
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
-  def removeRecord(eori: String, recordId: String): Action[JsValue] =
-    (authAction(eori) andThen validateHeaderAction).async(parse.json) { implicit request =>
+
+  def removeRecord(eori: String, recordId: String, actorId: String): Action[AnyContent] =
+    (authAction(eori) andThen validateHeaderAction).async { implicit request =>
       (for {
-        updateRecordRequest <- validateBody(request)
-        validRecordId       <- validateRecordId(recordId)
-        response            <- sendRemove(eori, validRecordId, updateRecordRequest)
-      } yield Ok(Json.toJson(response))).merge
+        validRecordId <- validateRecordId(recordId)
+        validActorId  <- validateActorId(actorId)
+        response      <- sendRemove(eori, validRecordId, validActorId)
+      } yield response).value.map {
+        case Right(_)          => NoContent
+        case Left(errorResult) => errorResult
+      }
     }
 
-  private def validateBody(request: Request[JsValue]): EitherT[Future, Result, RemoveRecordRequest] =
-    EitherT
-      .fromEither[Future](validateRequestBody[RemoveRecordRequest](request.body, uuidService))
-      .leftMap(r => BadRequest(Json.toJson(r)))
-
   def validateRecordId(recordId: String): EitherT[Future, Result, String] = {
```

Comment:
@mauro-fanton I have refactored the method as you suggested but I specified the return type instead of using a wildcard. 

`EitherT[Future, Result, String]`

Because `recordId` is converted from UUID to string

Can you check again please?